syncer/db.go

```diff
@@ -143,6 +150,12 @@ func (conn *DBConn) resetConn(tctx *tcontext.Context) error {
 	if err != nil {
 		return err
 	}
+	if conn.baseConn != nil {
+		err := conn.baseConn.Close()
```

Comment:
should we need to remove the conn in `BaseDB.conns`.

now, we may close a conn in or out of the `BaseDB`.

Reply:
I add a new function `CloseBaseConn` in `BaseDB`, and put it into [resetFunction](https://github.com/pingcap/dm/pull/266/files#diff-c3cc6fccd12f8e1679a575d1aaf8fd14R274) before we get a new `BaseConn`, I think it's better to controll connection in `BaseDB`, please take a look